src/components/EmbedMachine/index.tsx

```diff
@@ -1,21 +1,37 @@
 import React from 'react';
 import styles from './styles.module.css';
+import { useColorMode } from '@docusaurus/theme-common';
 
 type Embed = {
   name: string;
   embedURL: string;
 };
 
-export default function EmbedMachine({name, embedURL}: Embed) {
+export default function EmbedMachine({ name, embedURL }: Embed) {
+  const { colorMode } = useColorMode();
   return (
-    <p>
-        <iframe
-            loading="lazy"
-            src={embedURL}
-            className={styles.embed}
-        >
-            <a href={embedURL}>View the <em>{name}</em> machine in Stately Studio</a>.
-        </iframe>
-    </p>
+    <iframe
+      loading="lazy"
+      src={manageURL(embedURL, {
+        colorMode,
+      })}
+      className={styles.embed}
+    >
+      <a href={embedURL}>
+        View the <em>{name}</em> machine in Stately Studio
+      </a>
+      .
+    </iframe>
   );
-}
\ No newline at end of file
+}
+
+function manageURL(
+  embedURL: string,
+  options?: { colorMode: 'light' | 'dark' },
+): string {
+  const url = new URL(embedURL);
+  for (const opt in options) {
+    url.searchParams.set(opt, options[opt]);
+  }
+  return url.toString();
+}
```

Comment:
This function makes sure you don't have to think if the passed `embedURL` already contains a `colorMode` or not. When it goes through this function, the theme `colorMode` will replace it if exists.

Reply:
Amazing, thank you!